packages/formatter-html/src/result.ts

```diff
@@ -0,0 +1,239 @@
+            });
+
+            if (hint) {
+                this.cache.set(lowerCaseName, hint);
+            }
+        }
+
+        return hint;
+    }
+
+    public addHint(name: string): HintResult {
+        let hint = this.getHintByName(name);
+
+        if (hint) {
+            return hint;
+        }
+
+        hint = new HintResult(name, 'pass', this.url, this.isScanner);
+
+        this.hints.push(hint);
+
+        return hint;
+    }
+
+    public addProblem(problem: Problem) {
+        const hintId = problem.hintId;
+
+        let hint = this.getHintByName(hintId);
+
+        if (!hint) {
+            // All the problems have to have the same severity, so we just need to calculate the status once.
+            const status = problem.severity === Severity.error ? 'error' : 'warning';
+
+            hint = new HintResult(hintId, status, this.url, this.isScanner);
+
+            this.hints.push(hint);
+        }
+
+        if (problem.severity === Severity.error) {
+            this.errors++;
+        } else if (problem.severity === Severity.warning) {
+            this.warnings++;
+        }
+
+        hint.addProblem(problem);
+    }
+}
+
+export default class AnalysisResult {
+    public warnings: number;
+    public errors: number;
+    public scanTime: string;
+    public version?: string;
+    public permalink: string;
+    public categories: Array<Category>;
+    public url: string;
+    public isFinish: boolean;
+    public status: string;
+    public id: string;
+    public isScanner: boolean;
+    private cache: Map<string, Category> = new Map();
+
+    public constructor(target: string, options: FormatterOptions) {
+        this.url = target;
+        this.warnings = 0;
+        this.errors = 0;
+        this.status = options.status ? options.status : 'finished';
+        // Question: Should we have this here or in webhint.io?
+        this.isFinish = this.status === 'finished' || this.status === 'error';
+
+        this.scanTime = this.parseScanTime(options.scanTime || 0);
+        this.version = options.version;
+        this.permalink = '';
+        this.id = '';
+        this.isScanner = !!options.isScanner;
+
+        this.categories = [];
+    }
+
+    private pad = (timeString: string): string => {
+        return timeString && timeString.length === 1 ? `0${timeString}` : timeString;
+    };
+
+    private parseScanTime(scanTime: number): string {
+        const duration = moment.duration(scanTime);
+        const minutes = this.pad(`${duration.get('minutes')}`);
+        const seconds = this.pad(`${duration.get('seconds')}`);
+        let time = `${minutes}:${seconds}`;
+
+        if (duration.get('hours') > 0) {
+            const hours = this.pad(`${duration.get('hours')}`);
+
+            time = `${hours}:${time}`;
+        }
+
+        return time;
+    }
+
+    public getCategoryByName(name: string): Category | undefined {
+        const lowerCaseName = name.toLowerCase();
```

Comment:
`name` sometimes is `undefined`.

```
  hint:cli:analyze Failed to analyze: http://example.com/ +13s
  hint:cli:analyze TypeError: Cannot read property 'toLowerCase' of undefined
  hint:cli:analyze     at Result.getCategoryByName (/Users/xxxxx/projects/hint/packages/formatter-html/dist/src/result.js:118:36)
  hint:cli:analyze     at Result.addProblem (/Users/xxxxx/projects/hint/packages/formatter-html/dist/src/result.js:132:29)
  hint:cli:analyze     at problems.forEach (/Users/xxxxx/projects/hint/packages/formatter-html/dist/src/formatter.js:63:20)
  hint:cli:analyze     at Array.forEach (<anonymous>)
  hint:cli:analyze     at HTMLFormatter.format (/Users/xxxxx/projects/hint/packages/formatter-html/dist/src/formatter.js:62:18)
  hint:cli:analyze     at each (/Users/xxxxx/projects/tmp/node_modules/hint/dist/src/lib/cli/analyze.js:242:29)
  hint:cli:analyze     at /Users/xxxxx/projects/tmp/node_modules/hint/node_modules/async/dist/async.js:196:27
  hint:cli:analyze     at /Users/xxxxx/projects/tmp/node_modules/hint/node_modules/async/dist/async.js:74:12
  hint:cli:analyze     at /Users/xxxxx/projects/tmp/node_modules/hint/node_modules/async/dist/async.js:3110:16
  hint:cli:analyze     at eachOfArrayLike (/Users/xxxxx/projects/tmp/node_modules/hint/node_modules/async/dist/async.js:1069:9)
  hint:cli:analyze     at eachOf (/Users/xxxxx/projects/tmp/node_modules/hint/node_modules/async/dist/async.js:1117:5)
  hint:cli:analyze     at eachLimit (/Users/xxxxx/projects/tmp/node_modules/hint/node_modules/async/dist/async.js:3172:5)
  hint:cli:analyze     at eachLimit (internal/util.js:280:30)
  hint:cli:analyze     at print (/Users/xxxxx/projects/tmp/node_modules/hint/dist/src/lib/cli/analyze.js:241:15)
  hint:cli:analyze     at exports.default (/Users/xxxxx/projects/tmp/node_modules/hint/dist/src/lib/cli/analyze.js:253:19) +0ms
  hint:cli:analyze Total runtime: 11580ms +6ms
```

Reply:
I can't reproduce it :(

Reply:
So, the problem is because I was using the current version of `hint`, and this requires a new version of `hint`.